torchbenchmark/models/nvidia_deeprecommender/__init__.py

```diff
@@ -0,0 +1,88 @@
+import torch
+import torch.optim as optim
+import torchvision.models as models
+from ...util.model import BenchmarkModel
+from torchbenchmark.tasks import RECOMMENDATION
+
+import gc
+from .nvtrain import DeepRecommenderTrainBenchmark
+from .nvinfer import DeepRecommenderInferenceBenchmark
+
+class Model(BenchmarkModel):
+  
+  task = RECOMMENDATION.RECOMMENDATION
+  
+  def __init__(self, device="cpu", jit=False):
+    super().__init__()
+    self.devicename = device
+    self.notimplementedreason = "Implemented"
+    self.train_obj = DeepRecommenderTrainBenchmark(device = device, jit = jit)
+    self.infer_obj = DeepRecommenderInferenceBenchmark(device = device, jit = jit)
+    if jit:
+      self.notimplementedreason = "Jit Not Supported"
+    
+    elif self.devicename != "cpu" and self.devicename != "cuda":
+      self.notimplementedreason = "device type not supported"
+
+    self.evalMode = True
+
+  def get_module(self):
```

Comment:
once you change your eval() to invoke self.infer_obj.rencoder directly, you'll be able to just return self.infer_obj.rencoder here.  likewise for train mode, you can just return self.train_obj.model here, i think.  although i'm not sure if other benchmarks implemented get_module for both train and eval, or just eval.

Reply:
In my case I only return the train model, because the eval one just loads the model constructed from train.

Reply:
in many cases there will only be one model obj.  it's fine to just return that.